### PR TITLE
KAFKA-16302: Remove check for log message that is no longer present (fix builds)

### PR DESCRIPTION
**Pull request by jolshan**
https://github.com/apache/kafka/commit/a3528a316f4df969b9995f00af89161a5840a547 removed this log but not the test asserting it.

Builds are currently red because for some reason these tests can't retry. We should address that as a followup.

### CODE REVIEW

**Review by gharris1727 — COMMENTED** [review]
This whole appender block is unused now that log messages aren't being asserted on.

Can we revert the log message removal instead?
Or bring it back at TRACE level?

**Comment by gharris1727** [comment]
Thanks for root-causing this so quickly @jolshan I was just seeing it pop up in my PR builds.

**Comment by jolshan** [comment]
>This whole appender block is unused now that log messages aren't being asserted on.

Good point, I can remove

> Can we revert the log message removal instead?
Or bring it back at TRACE level?

The metric should be sufficient to test this. Not sure we need the log.

**Comment by gharris1727** [comment]
> The metric should be sufficient to test this. Not sure we need the log.

I didn't read the full test before commenting earlier. I think this is reasonable then.

The test name "shouldLogAndMeasureExpiredRecords" should probably be "shouldMeasureExpiredRecords" then.

**Review by mjsax — APPROVED** [review]
Thanks for the fix!

**Review by gharris1727 — APPROVED** [review]
Tests pass for me locally. IMHO this can be merged without waiting for the Jenkins build because it will unblock other PRs, but I'll leave that up to you.

**Comment by jolshan** [comment]
I'll run streams locally as a double check and then merge 👍

**Pull request merged**
